nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/script/impl/JythonScriptEngineConfigurator.java

```diff
@@ -41,25 +48,27 @@ public String getScriptEngineName() {
     }
 
     @Override
-    public Object init(ScriptEngine engine, String[] modulePaths) throws ScriptException {
-        if (engine != null) {
-            // Need to import the module path inside the engine, in order to pick up
-            // other Python/Jython modules.
-            engine.eval("import sys");
-            if (modulePaths != null) {
-                for (String modulePath : modulePaths) {
-                    engine.eval("sys.path.append(" + PyString.encode_UnicodeEscape(modulePath, true) + ")");
-                }
-            }
-        }
+    public Object init(ScriptEngine engine, String[] modulePaths) {
```

Comment:
Previously, the init() method evaluated the script. Which means that if the script doesn't compile, the processor would be invalid. Now, with the init() method not compiling the script, does that mean that the processor will be valid even if the script is not compilable? Might make sense to compile in the init() method instead of the first iteration of eval?